Dockerfile

```diff
@@ -1,22 +1,42 @@
-FROM node:alpine AS base
-RUN mkdir -p /code
-WORKDIR /code
-CMD "hsd"
+FROM node:10-alpine AS base
 
-RUN apk upgrade --no-cache && \
-    apk add --no-cache bash unbound-dev gmp-dev
+RUN mkdir /opt/hsd
+WORKDIR /opt/hsd
 
-COPY package.json \
-     #package-lock.json \
-     /code/
+# dynamically linked deps
+RUN apk upgrade --no-cache \
+  && apk add --no-cache unbound-dev
 
-# Install build dependencies and compile
 FROM base AS build
-RUN apk add --no-cache g++ gcc make python2
-RUN npm install --production
+
+# build deps
+RUN apk upgrade --no-cache \
```

Comment:
Again, would remove the `apk upgrade` command from here as well.  If I'm being nitpicky, the packages being installed below should also have versions pinned to them, but these packages are pretty solid and I wouldn't expect anything out of the ordinary.